core/src/main/java/com/graphhopper/reader/dem/TunnelElevationInterpolator.java

```diff
@@ -29,12 +29,13 @@
  */
 public class TunnelElevationInterpolator extends AbstractEdgeElevationInterpolator {
 
-    public TunnelElevationInterpolator(GraphHopperStorage storage, DataFlagEncoder dataFlagEncoder) {
-        super(storage, dataFlagEncoder);
+    public TunnelElevationInterpolator(GraphHopperStorage storage, StringEncodedValue roadEnvironmentEnc) {
+        super(storage, roadEnvironmentEnc);
     }
 
     @Override
     protected boolean isInterpolatableEdge(EdgeIteratorState edge) {
-        return dataFlagEncoder.isTunnel(edge.getData());
+        //TODO We should not hardcode this string here
+        return edge.get(roadEnvironmentEnc).equals("tunnel");
```

Comment:
For now I just hardcoded this. But we should have a way to access these properties.

Reply:
The way it should work is something like

```java
constructor {
   tunnelIndex = roadEnvironmentEnc.indexOf("tunnel")
   // is this too ugly?
   roadEnvironmentEnc = (IntEncodedValue) roadEnvironmentEnc;
}
isInterpolatableEdge {
   return edge.get(roadEnvironmentEnc) == tunnelIndex;
}
```

Reply:
Ah yes makes sense.

Reply:
> The way it should work is something like

Could you explain why? The roadEnvironmentEnc is a StringEncodedValue. The get method would therefore return a String? The tests also succeeded like that.

Or was your comment about having a short cut and saving some computation time by not getting the stored string but the index? This would then require a new method for the EdgeIteratorState, like `getRaw`?

Reply:
If you convert it into an IntEncodedValue the raw integer is returned (as `edge.get(IntEncodedValue)` is called) and no expensive string comparison is necessary and so it can be used e.g. in the calcWeight method.

> Or was your comment about

The comment regarding 'too ugly' is about the cast and so this usage with better perf is a bit too hidden.

>  This would then require a new method for the EdgeIteratorState, like getRaw?

I would not introduce a new method for EdgeIteratorState, instead we could think about roadEnvironmentEnc.getIntEncodedValue instead of the cast?

Reply:
The easiest solution I found was to simply cast the StringEncodedValue to IntEncodedValue. I just pushed that change. I haven't measured performance, but I think the cast should be alright?